packages/jaeger-ui/src/components/SearchTracePage/SearchResults/ResultItem.tsx

```diff
@@ -38,10 +40,43 @@ type Props = {
   disableComparision: boolean;
 };
 
+type State = {
+  erroredServices: Set<string>;
+  numSpans: number;
+  numErredSpans: number;
+  timeStr: string;
+  fromNow: string | boolean;
+};
+
 const isErrorTag = ({ key, value }: KeyValuePair) => key === 'error' && (value === true || value === 'true');
 const trackTraceConversions = () => trackConversions(EAltViewActions.Traces);
 
-export default class ResultItem extends React.PureComponent<Props> {
+export default class ResultItem extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    const { startTime, spans } = props.trace;
+
+    const mDate = moment(startTime / 1000);
+
+    const erroredServices: Set<string> = new Set<string>();
+
+    const numErredSpans = spans.filter(sp => {
+      const hasError = sp.tags.some(isErrorTag);
+      if (hasError) {
+        erroredServices.add(sp.process.serviceName);
+      }
+      return hasError;
+    }).length;
+
+    this.state = {
```

Comment:
just curious, how/where is this `.state` field associated with the `State` type?

Reply:
```
export default class ResultItem extends React.PureComponent<Props, State> {
```
It is associated here :)